In [1]:
# Import pandas
import pandas as pd

In [2]:
# Create a DataFrame with balance.txt
df = pd.read_csv('balance.txt', delim_whitespace=True)

In [3]:
# Compare average 'Income' based on 'Ethnicity'.
ethnicity_average_income = df.groupby('Ethnicity')['Income'].mean()
print(
    "Average income by Ethnicity:"
    "\n\n"
    f"African American: £{ethnicity_average_income['African American']:.2f}"
    "\n"
    f"Asian: £{ethnicity_average_income['Asian']:.2f}"
    "\n"
    f"Caucasian: £{ethnicity_average_income['Caucasian']:.2f}"
    )

Average income by Ethnicity:

African American: £47.68
Asian: £44.19
Caucasian: £44.52


When comparing average income by ethnicity, we can see that individuals of African American (£47.68) ethnicity earn more income on average than individuals of Asian (£44.19) and Caucasian (£44.52) ethnicities.

In [7]:
# Compare average income based on marital status ('Married').
married_average_income = df.groupby('Married')['Income'].mean()
print(
    "Average income by Marital Status:"
    "\n\n"
    f"Married: £{married_average_income['Yes']:.2f}"
    "\n"
    f"Unmarried: £{married_average_income['No']:.2f}"
    )

Average income by Marital Status:

Married: £46.22
Unmarried: £43.64


On average, we see that married individuals (£46.22) earn more income than unmarried individuals (£43.64).

In [8]:
# Highest income in DataFrame.
highest_income = df.loc[:, 'Income'].max()
print(f"Highest Income: £{highest_income:.2f}")

Highest Income: £186.63


The highest income within our DataFrame is £186.63.

In [9]:
# Lowest income in DataFrame.
lowest_income = df.loc[:, 'Income'].min()
print(f"Lowest Income: £{lowest_income:.2f}")

Lowest Income: £10.35


The lowest income within our DataFrame is £10.35.

In [10]:
# Total number of 'Cards' wihtin DataFrame.
total_cards = df.loc[:, 'Cards'].sum()
print(f"Total Cards: {total_cards}")

Total Cards: 1183


The total number of cards held by all individuals within our DataFrame is 1183.

In [11]:
# Group the DataFrame by the 'Gender' column, count the non-null 'Age' values for each gender,
# and store the resulting Series in gender_count.
gender_count = df.groupby('Gender')['Age'].count()
# Use the indices of the above Series to display the information in a readable format.
print(
    "Number of individuals by gender:"
    "\n\n"
    f"Female: {gender_count['Female']}"
    "\n"
    f"Male: {gender_count['Male']}"
    )

Number of individuals by gender:

Female: 207
Male: 193


Our DataFrame consists of 207 female individuals and 193 male individuals.